lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -23,6 +23,18 @@
 using namespace circt;
 using namespace firrtl;
 
+//===----------------------------------------------------------------------===//
+// Helpers
+//===----------------------------------------------------------------------===//
+
+static bool isAncestor(Operation *op, Value value) {
+  if (auto result = dyn_cast<OpResult>(value))
+    return op->isAncestor(result.getOwner());
+  if (auto argument = dyn_cast<BlockArgument>(value))
+    return op->isAncestor(argument.getOwner()->getParentOp());
+  return false;
```

Comment:
Could this be a blind cast (which would pull in an assert for debug-mode checking)? (Can this be anything else?)